agent/src/main.rs

```diff
@@ -536,6 +538,10 @@ fn apply_smf(
 
     for (_, region_snapshots) in running_snapshots.iter_mut() {
         for snapshot in region_snapshots.values_mut() {
+            if snapshot.state != State::Requested {
```

Comment:
Stupid github.  I want to comment on lines 370 and 380 in this file, but github won't let me.
So, I'm putting the comment on this line even though it is not related to the line itself.

Just FYI though.  I thought those lines might have been where we were leaking memory in 
agent. (relating to https://github.com/oxidecomputer/crucible/issues/542)

Reply:
You mean

```rust
                /*
                 * XXX just disable for now.
                 */
                inst.disable(false)?;
```